wandb/apis/workspaces/interface.py

```diff
@@ -0,0 +1,449 @@
+
+from . import expr, internal
+
+dataclass_config = ConfigDict(validate_assignment=True, extra="forbid", slots=True)
+
+
+def is_not_all_none(v):
+    if v is None or v == "":
+        return False
+    if isinstance(v, Iterable) and not isinstance(v, str):
+        return any(v not in (None, "") for v in v)
+    return True
+
+
+def is_not_internal(k):
+    return not k.startswith("_")
+
+
+def should_show(v):
+    """This is a workaround because BaseMetric.__eq__ returns FilterExpr."""
+    if isinstance(v, Iterable) and not isinstance(v, str):
+        return any(should_show(x) for x in v)
+    if isinstance(v, expr.BaseMetric):
+        return True
+    return False
+
+
+@dataclass(config=dataclass_config, repr=False)
+class Base:
+    def __repr__(self):
+        fields = (
+            f"{k}={v!r}"
+            for k, v in self.__dict__.items()
+            if (is_not_all_none(v) and is_not_internal(k)) or (should_show(v))
+        )
+        fields_str = ", ".join(fields)
+        return f"{self.__class__.__name__}({fields_str})"
+
+    def __rich_repr__(self):
+        for k, v in self.__dict__.items():
+            if (is_not_all_none(v) and is_not_internal(k)) or (should_show(v)):
+                yield k, v
+
+    @property
+    def _model(self):
+        return self.to_model()
+
+    @property
+    def _spec(self):
+        return self._model.model_dump(by_alias=True, exclude_none=True)
+
+
+@dataclass
+class SectionPanelSettings(Base):
+    """Settings for a panels section, typically seen at the top right of the section in the UI.
+
+    Args:
+        layout (str): The layout of the panels in the section.  "standard" follows the grid layout, while "custom" allows for per-panel layouts.
+        columns (int): The number of columns in the layout.
+        rows (int): The number of rows in the layout.
+
+    """
+
+    layout: Literal["standard", "custom"] = "standard"
+    columns: int = 3
+    rows: int = 2
+
+    @classmethod
+    def from_model(cls, model: internal.FlowConfig):
+        return cls(
+            layout="standard" if model.snap_to_columns else "custom",
+            columns=model.columns_per_page,
+            rows=model.rows_per_page,
+        )
+
+    def to_model(self):
+        return internal.FlowConfig(
+            snap_to_columns=self.layout == "standard",
+            columns_per_page=self.columns,
+            rows_per_page=self.rows,
+        )
+
+
+@dataclass(config=dataclass_config, repr=False)
+class Section(Base):
+    """Represents a section in a workspace.
+
+    Args:
+        name (str): The name of the section.
+        panels (LList[PanelTypes]): A list of panels in the section.
+        collapsed (bool): Whether the section is collapsed.
+        section_panel_settings (SectionPanelSettings): Settings for the panels in this section.
+    """
+
+    name: str
+    panels: LList[PanelTypes] = Field(default_factory=list)
+    collapsed: bool = False
```

Comment:
Can we try to match the view spec field and reverse it to be positive logic? 
```suggestion
    is_open: bool = True
```